RFCs/FS-1142-Extended-numeric-literal.md

```diff
@@ -0,0 +1,137 @@
+    let x6 = 0x_52            // OK (Invalid *previously*)
+    let x7 = 0x5_2            // OK (hexadecimal literal)
+    let x8 = 0x52_            // Invalid cannot put underscores at the end of a number
+
+    // In contrast to Java, literals with leading zeros are decimal in F#.
+    let x9 = 0_52             // OK (decimal literal)
+    let x10 = 05_2            // OK (decimal literal)
+    let x11 = 052_            // Invalid cannot put underscores at the end of a number
+
+    // To create an octal literal, prefix it with '0o' similar to hexadezimal literals. The same rules apply:
+    let x12 = 0_o52            // Invalid cannot put underscores in the 0o radix prefix
+    let x13 = 0o_52            // OK (Invalid *previously*)
+    let x14 = 0o5_2            // OK (octal literal)
+    let x15 = 0o52_            // Invalid cannot put underscores at the end of a number
+    ```
+
+2. Allow number prefix (`0x`, `0o`, `0b`) before integer custom numeric literals.
+  
+    ```fsharp
+    let x1 = 0x123I        // big int (291)
+    let x2 = 0o123I        // big int (83)
+    let x3 = 0b1010I       // big int (10)
+    ```
+
+    The number prefix **will not be removed** from the string sended to the `FromString` of the numeric literal module. This may be a break change.
+
+    ```fsharp
+    module NumericLiteralG =
+      let FromString s = s
+
+    let x4 = 0x_99_9999_9999_9999_9999G    // x4 will be "0x999999999999999999"
+    ```
+
+3. Allow floating point custom numeric literals. 
+
+    This will require the numeric literal module contains two new functions:
+
+    ```fsharp
+    val FromFloat: float -> 'CustomNumber
+    val FromFloatString: string -> 'CustomNumber
+    ```
+
+    According to [this comment](https://github.com/fsharp/fslang-suggestions/issues/445#issuecomment-596902041),
+
+    - When the literal has <= 15 significant figures[^1] and its exponent is within -300 to 300, the compiler will call `FromFloat` with the parsed `float` number
+    - Or the compiler will call `FromFloatString` with the original string
+
+    [^1]: The first none zero figure to last none zero figure, without the dot. Can be match by the Regex:
+
+        ```regex
+        ^-?0*(?<number>\d+\.?\d*?)0*(?:$|[eE][+-]?(?<exp>\d+))
+        ```
+
+    ```fsharp
+    module NumericLiteralG =
+      let FromFloat (s: float) = s
+      let FromFloatString (s: string) = s
+
+    let x1 = 123.456e-10G    // x1: float = 1.23456e-08
+    let x2 = 123.456789123456789123e-10G    // x2: string = "123.456789123456789123e-10"
+    ```
+
+# Drawbacks
+
+- Custom numeric literals module might harder to write.
+
+- May introduce break change.
+
+# Alternatives
+
+- For number prefix (`0x`, `0o`, `0b`) before integer custom numeric literals, we might introduce a new `FromIntegerString` to avoid the break change.
+- Or firstly parse it to `bigint` then `ToString` to obtain a literal without prefix.
```

Comment:
This is a much better suggestion. In fact, you don't even have to do that. Conversion is rather trivial and can be done on-the-fly. 

This would certainly be my preference and would avoid any burden on people writing custom numerals modules.